src/plugins/discover/public/application/main/hooks/utils/build_state_subscribe.ts

```diff
@@ -47,15 +47,29 @@ export const buildStateSubscribe =
     const nextQuery = nextState.query;
     const savedSearch = savedSearchState.getState();
     const prevQuery = savedSearch.searchSource.getField('query');
+    const isTextBasedQueryLang = isTextBasedQuery(nextQuery);
     const queryChanged = !isEqual(nextQuery, prevQuery) || !isEqual(nextQuery, prevState.query);
+    // check if prevState and nextState are equal except for index
+
+    if (
+      isTextBasedQueryLang &&
+      (nextState.index === undefined || nextState.viewMode === undefined) &&
+      !queryChanged
+    ) {
+      // this is just used to unset the index/viewmode when switching to a text based query, nothing to do here
+      addLog('[appstate] subscribe update ignored for text base ql', {
+        prevState,
+        nextState,
+      });
+      return;
```

Comment:
this is part of the function that's triggered when the appState (which also syncs with the URL) changes, for ES|QL it means that the execution of this function is stopped, when the state ist just cleaned up (index or viewMode set to undefined) by the `useTextBasedQueryLanguage` hook, and no query change is involved